contracts/JBController.sol

```diff
@@ -830,15 +893,43 @@ contract JBController is IJBController, JBTerminalUtility, JBOperatable, Reentra
           _constraints.terminal
         ] = _constraints.distributionLimit;
 
-      // Set the overflow allowance if there is one.
-      if (_constraints.overflowAllowance > 0)
-        overflowAllowanceOf[_projectId][_fundingCycle.configuration][
-          _constraints.terminal
-        ] = _constraints.overflowAllowance;
+      // Get a reference to the overflow allowance data for this project and this terminal.
+      JBConfigurationOverflowAllowanceData[]
+        memory _overflowAllowanceData = _overflowAllowanceDataOf[_projectId][_constraints.terminal];
+
+      // Check if _constraints.overflowAllowance exists and differs from latest value.
+      // If so, push the value into the array.
+      // This prevents needlessly storing overflow allowances for each new reconfiguration if it isn't changing.
+      if (
+        (_constraints.overflowAllowance > 0 && _overflowAllowanceData.length == 0) ||
+        _overflowAllowanceData[_overflowAllowanceData.length].overflowAllowance !=
+        _constraints.overflowAllowance
+      )
+        _overflowAllowanceDataOf[_projectId][_constraints.terminal].push(
+          JBConfigurationOverflowAllowanceData(
+            uint56(_fundingCycle.configuration),
+            _constraints.overflowAllowance
+          )
+        );
 
-      if (_constraints.currency > 0)
-        currencyOf[_projectId][_fundingCycle.configuration][_constraints.terminal] = _constraints
-          .currency;
+      // Get a reference to the currency data for this project and this terminal.
+      JBConfigurationCurrencyData[] memory _currencyData = _currencyDataOf[_projectId][
+        _constraints.terminal
+      ];
+
+      // Check if _constraints.currency exists and differs from latest value.
+      // If so, push the value into the array.
+      // This prevents needlessly storing currencies for each new reconfiguration if it isn't changing.
+      if (
+        (_constraints.currency > 0 && _currencyData.length == 0) ||
+        _currencyData[_currencyData.length].currency != _constraints.currency
+      )
+        _currencyDataOf[_projectId][_constraints.terminal].push(
```

Comment:
Very curious about the actual cost diff of pushing to an array infrequently vs. updating a map value everytime